library/ssl_12_cli.c

```diff
@@ -1,5 +1,6 @@
 /*
- *  SSLv3/TLSv1 client-side functions
+ *  Client-specific handshake reading/writing functions
+ *  for TLS versions <= 1.2.
```

Comment:
The `_12` files contain support not only for TLS 1.2 but also for older versions. I find the name misleading: in `ssl_12*`, I expected to find code that is specific to TLS 1.2, not code for TLS 1.1 and 1.0 and SSL 3.0.

Reply:
I'm open for suggestions.

Reply:
Perhaps `ssl_30-12*.c` or `ssl_30_tls_12*.c`, or `ssl_30_to_tls_12*.c` or `ssl_up_to_tls12*.c`?

And for TLS 1.3 files, perhaps `ssl_tls13*.c`?

Btw, I'm not sure I'm entirely happy with `_gen` either for "generic", as it could also stand for "generated" or "generate". It's obviously a big improvement over the current "tls" which hardly gives a hint, but I'm thinking perhaps `_common` would be better? And while at it, why not change `_cli` and `_srv` to `_client` and `_server`?

Reply:
As mentioned below, I prefer to keep the splitting minimal, solely achieving the goal of separating existing protocol logic from messaging logic. I'm fine with splitting further, but I prefer this to be done in a separate PR.